spark/src/test/java/org/apache/iceberg/spark/source/TestSparkTableUtil.java

```diff
@@ -199,4 +204,41 @@ public void testImportAsHiveTable() throws Exception {
     long count2 = spark.read().format("iceberg").load(DB_NAME + ".test_partitioned_table").count();
     Assert.assertEquals("three values ", 3, count2);
   }
+
+  @Test
+  public void testImportWithIncompatibleSchema() throws Exception {
+    spark.table(qualifiedTableName).write().mode("overwrite").format("parquet")
+        .saveAsTable("original_table");
+
+    // The field is different so that it will project with name mapping
```

Comment:
Name mapping should only be used if it is set on the table.

Reply:
Done.